tests/unit/clients/workflow_test.py

```diff
@@ -17,6 +17,23 @@ def workflow_config():
 
 
 @pytest.fixture
+def boto_client():
```

Comment:
Let's not overwrite the base boto_client with return values. Defining the return values for this mock at the test level is more explicit and lessens global assumptions about this mock. If you want to DRY the return values, you could rename this to something else.